yaml_specs/modulemd_stream_v2.yaml

```diff
@@ -4,40 +4,66 @@ document: modulemd
 # Module metadata format version
 version: 2
 data:
-    # Module name, optional
-    # Typically filled in by the buildsystem, using the VCS repository
-    # name as the name of the module.
+    # Module name
+    # Filled in by the build system, using the VCS repository name as the name
+    # of the module.
+    #
+    # When included as input to the build system, the behavior is unspecified.
+    #
+    # Mandatory for module metadata in a yum/dnf repository.
     name: foo
-    # Module update stream, optional
-    # Typically filled in by the buildsystem, using the VCS branch name
-    # as the name of the stream.
+
+    # Module update stream
+    # Filled in by the buildsystem, using the VCS branch name as the name of
+    # the stream.
+    #
+    # When included as input to the build system, the behavior is unspecified.
+    #
+    # Mandatory for module metadata in a yum/dnf repository.
     stream: latest
-    # Module version, integer, optional, cannot be negative
-    # Typically filled in by the buildsystem, using the VCS commit
-    # timestamp.  Module version defines upgrade path for the particular
-    # update stream.
+
+    # Module version, integer, cannot be negative
+    # Filled in by the buildsystem, using the VCS commit timestamp.  Module
+    # version defines upgrade path for the particular update stream.
+    #
+    # When included as input to the build system, the behavior is unspecified.
+    #
+    # Mandatory for module metadata in a yum/dnf repository.
     version: 20160927144203
-    # Module context flag, optional
+
+    # Module context flag
     # The context flag serves to distinguish module builds with the
     # same name, stream and version and plays an important role in
     # future automatic module stream name expansion.
     # Filled in by the buildsystem.  A short hash of the module's name,
-    # stream, version and its expanded runtime dependencies.
+    # stream, version and its expanded runtime dependencies. The exact
+    # mechanism of generating the has is unspecified except that it must be
+    # stable.
+    # When included as input to the build system, the behavior is unspecified.
```

Comment:
Please empty comment lines above and below for clarity as was done for `name`, `stream`, and `version`.